ICMP checksum calculation produces segmentation fault

After recent changes to ICMP checksum calculation NAT example crashes when pinged:
```
DEBUG: ---------------
DEBUG: System is using 7 cores now. 1 cores are left available.
DEBUG: Number of packets in queue for split:  0
DEBUG: Current speed of splitter 2 is 0 PKT/S
DEBUG: Number of packets in queue for split:  0
DEBUG: Current speed of splitter 4 is 0 PKT/S
DEBUG: Mempool N 0 used 128 from 32764
DEBUG: Mempool N 1 used 128 from 32764
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5056ca]

goroutine 57 [running, locked to thread]:
github.com/intel-go/yanff/packet.calculateDataChecksum(0x0, 0x38, 0x0, 0xc42072a000)
        /home/ubuntu/go/src/github.com/intel-go/yanff/packet/checksum.go:148 +0x1a
github.com/intel-go/yanff/packet.CalculateIPv4ICMPChecksum(0x7fa415c0134e, 0x7fa415c01362, 0x0, 0x7fa48c0091e0)
        /home/ubuntu/go/src/github.com/intel-go/yanff/packet/checksum.go:292 +0x4c
github.com/intel-go/yanff/examples/nat.setIPv4ICMPChecksum(0x7fa415c012c0, 0xc420720101)
        /home/ubuntu/go/src/github.com/intel-go/yanff/examples/nat/cksum.go:68 +0xaf
github.com/intel-go/yanff/examples/nat.(*ipv4Port).handleICMP(0xc4200a63c0, 0x7fa415c012c0, 0x0)
        /home/ubuntu/go/src/github.com/intel-go/yanff/examples/nat/translation.go:366 +0xf4
github.com/intel-go/yanff/examples/nat.PublicToPrivateTranslation(0x7fa415c012c0, 0xa7d0c0, 0xb7eda0, 0xa7d0c0)
        /home/ubuntu/go/src/github.com/intel-go/yanff/examples/nat/translation.go:95 +0x6ce
github.com/intel-go/yanff/flow.split(0x703f00, 0xc42007c4e0, 0xc42009e1e0, 0xc4200b8000, 0x7fa49dc772d0, 0xb7eda0)
        /home/ubuntu/go/src/github.com/intel-go/yanff/flow/flow.go:1183 +0x5cd
github.com/intel-go/yanff/scheduler.(*Scheduler).startClonable.func1(0xc42005e0b0, 0x5)
        /home/ubuntu/go/src/github.com/intel-go/yanff/scheduler/scheduler.go:201 +0xe7
created by github.com/intel-go/yanff/scheduler.(*Scheduler).startClonable
        /home/ubuntu/go/src/github.com/intel-go/yanff/scheduler/scheduler.go:196 +0x1cc
```
You can use https://github.com/intel-go/yanff/wiki/NAT-example instruction to run NAT example.

--- a/examples/nat/cksum.go
+++ b/examples/nat/cksum.go
@@ -65,6 +65,7 @@ func setIPv4ICMPChecksum(pkt *packet.Packet, calculateChecksum, hWTXChecksum boo
 		}
 		l4 := pkt.GetICMPForIPv4NoCheck()
 		l4.Cksum = 0
-		l4.Cksum = packet.SwapBytesUint16(packet.CalculateIPv4ICMPChecksum(l3, l4, pkt.Data))
+		l4.Cksum = packet.SwapBytesUint16(packet.CalculateIPv4ICMPChecksum(l3, l4,
+			unsafe.Pointer(uintptr(unsafe.Pointer(l4))+common.ICMPLen)))
 	}
 }

